crates/ruff_cli/src/diagnostics.rs

```diff
@@ -401,8 +435,19 @@ pub(crate) fn lint_stdin(
     noqa: flags::Noqa,
     autofix: flags::FixMode,
 ) -> Result<Diagnostics> {
-    let mut source_kind = SourceKind::Python(contents.to_string());
-    let source_type = PySourceType::default();
+    let source_type = path.map(PySourceType::from).unwrap_or_default();
+
+    let mut source_kind = if source_type.is_jupyter() {
+        // SAFETY: Jupyter isn't the default type, so we must have a path.
+        match notebook_from_contents(contents, path.unwrap()) {
```

Comment:
Could we instead pass an `Option<Path>` to `notebook_from_contents`? This feels like something that's easy to get wrong in the future.

Reply:
We need it to format the error message. I could restructure the code such that we only take this branch if we have a Path though.